front/app/containers/Admin/formBuilder/components/FormBuilderSettings/index.tsx

```diff
@@ -46,18 +48,73 @@ const FormBuilderSettings = ({
 }: Props) => {
   const locales = useAppConfigurationLocales();
   const [currentTab, setCurrentTab] = useState<'content' | 'logic'>('content');
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [fieldIndexToDelete, setFieldIndexToDelete] = useState<number>();
   const { formatMessage } = useIntl();
-  const { watch } = useFormContext();
+  const { watch, setValue } = useFormContext();
+  const formCustomFields: IFlatCustomField[] = watch('customFields');
+
+  const closeModal = () => {
+    setShowDeleteModal(false);
+  };
+  const openModal = () => {
+    setShowDeleteModal(true);
+  };
 
   if (isNilOrError(locales)) {
     return null;
   }
 
+  const preDeletion = (fieldIndex: number) => {
+    let pageIsLinked = false;
+    setFieldIndexToDelete(fieldIndex);
+
+    // Check if deleted field has linked logic
+    formCustomFields.map((surveyField) => {
+      if (surveyField.logic && surveyField.logic.rules) {
+        const hasLinkedRule = surveyField.logic.rules.find(
+          (rule) =>
+            rule.goto_page_id === field.id ||
+            rule.goto_page_id === field.temp_id
+        );
+        if (hasLinkedRule) {
+          pageIsLinked = true;
+        }
+      }
+    });
+
+    // Open confirmation modal if linked logic present
+    if (pageIsLinked) {
+      openModal();
+    }
+
+    // Otherwise, delete the field without confirmation
+    if (!pageIsLinked) {
+      onDelete(fieldIndex);
+    }
```

Comment:
Actually, from workshop we had on Tuesday, Brent said that the proper flow when working on a feature branch is to not translate them in this branch, but wait until it's merged into the base (CL-1893-logic), and then do translations on the base  branch instead :) So that's what I was planning to do here. Arguably it might take an extra day though to get the translations in once it's merged, but I guess that's something to chat with the FE devs about :thinking: